mleap-xgboost-runtime/src/test/scala/ml/combust/mleap/xgboost/runtime/testing/FloatingPointApproximations.scala

```diff
@@ -0,0 +1,24 @@
+package ml.combust.mleap.xgboost.runtime.testing
+
+
+trait FloatingPointApproximations {
+
+  val DefaultMinimumPrecision = 1e-7
+
+  def almostEqual(x: Double, y: Double, precision: Double = DefaultMinimumPrecision): Boolean = {
+    if ((x - y).abs < precision) true else false
+  }
+
+  def almostEqualSequences(
```

Comment:
Is it really necessary to implement this ourselves? Is there no implementation in a library somewhere?

Reply:
We looked into it last quarter and haven't found anything. This is the same code we currently use internally to check float equality to the 1e-7